homeassistant/components/balboa/climate.py

```diff
@@ -0,0 +1,208 @@
+    def fan_mode(self) -> str:
+        """Return the current fan mode."""
+        fanmode = self._client.get_blower()
+        if fanmode is None:
+            return FAN_OFF
+        if fanmode == self._client.BLOWER_OFF:
+            return FAN_OFF
+        if fanmode == self._client.BLOWER_LOW:
+            return FAN_LOW
+        if fanmode == self._client.BLOWER_MEDIUM:
+            return FAN_MEDIUM
+        if fanmode == self._client.BLOWER_HIGH:
+            return FAN_HIGH
+        return FAN_OFF
+
+    @property
+    def precision(self) -> float:
+        """Return the precision of the system.
+
+        Balboa spas return data in C or F depending on how the display is set,
+        because ultimately, we are just reading the display.
+        In C, we have half-degree accuracy, in F, whole degree.
+        """
+        if self.hass.config.units.temperature_unit == TEMP_CELSIUS:
+            return PRECISION_HALVES
+        return PRECISION_WHOLE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement, as defined by the API."""
+        tscale = self._client.get_tempscale()
+        if tscale == self._client.TSCALE_C:
+            return TEMP_CELSIUS
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._client.get_curtemp()
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature we try to reach."""
+        return self._client.get_settemp()
+
+    @property
+    def min_temp(self) -> int:
+        """Return the minimum temperature supported by the spa."""
+        trange = self._client.get_temprange()
+        scale = self._client.get_tempscale()
+        return self._client.tmin[trange][scale]
+
+    @property
+    def max_temp(self) -> int:
+        """Return the minimum temperature supported by the spa."""
+        trange = self._client.get_temprange()
+        scale = self._client.get_tempscale()
+        return self._client.tmax[trange][scale]
+
+    @property
+    def preset_modes(self):
+        """Return the valid preset modes."""
+        modes = [
+            mode
+            for mode in self._client.get_heatmode_stringlist()
+            # only return "Ready in Rest" as an option if the spa is currently in that
+            # mode since it is a status rather than an available mode to be selected
+            if self._client.get_heatmode() == self._client.HEATMODE_RNR
+            or mode != self._client.get_heatmode_stringlist()[self._client.HEATMODE_RNR]
+        ]
+        return modes
+
+    @property
+    def preset_mode(self):
+        """Return current preset mode."""
+        return self._client.get_heatmode(True)
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            "time": f"{self._client.time_hour:02d}:{self._client.time_minute:02d}",
+        }
```

Comment:
We usually don't store the current time in the state machine since it will cause the state to be saved in the database even when it doesn't change because the time will increment

Reply:
So should I get rid of this extra attribute entirely then?  From my perspective it's mildly useful to be able to see the time that the spa is displaying on the clock, but if it's just too much db churn I can remove it entirely.

Reply:
Let's drop it.